pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -632,8 +639,56 @@ func ExpirationDateExpired(timestamp *metav1.Time) bool {
 	return time.Now().UTC().After(timestamp.Time) || time.Now().UTC().Equal(timestamp.Time)
 }
 
+// disablePodSecurityPolicyAdmissionController disables the PodSecurityPolicy Admission Controller of a shoot
+func disablePodSecurityPolicyAdmissionController(shoot *gardencorev1beta1.Shoot, reason string) []string {
+	var (
+		reasonsForUpdate []string
+	)
+
+	if shoot.Spec.Kubernetes.KubeAPIServer == nil {
+		shoot.Spec.Kubernetes.KubeAPIServer = &gardencorev1beta1.KubeAPIServerConfig{}
+	}
+
+	for i, admissionPlugin := range shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins {
+		if admissionPlugin.Name == "PodSecurityPolicy" {
+			if !pointer.BoolDeref(admissionPlugin.Disabled, false) {
+				shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins[i].Disabled = pointer.Bool(true)
+				reasonsForUpdate = append(reasonsForUpdate, reason)
+			}
+			return reasonsForUpdate
+		}
+	}
+
+	disabledAdmissionPlugin := gardencorev1beta1.AdmissionPlugin{
+		Name:     "PodSecurityPolicy",
+		Disabled: pointer.Bool(true),
+	}
+	shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins = append(shoot.Spec.Kubernetes.KubeAPIServer.AdmissionPlugins, disabledAdmissionPlugin)
+	reasonsForUpdate = append(reasonsForUpdate, reason)
+
+	return reasonsForUpdate
+}
+
+// ensureSufficientMaxWorkers ensures that the number of max workers of a worker group is greater or equal to its number of zones
+func ensureSufficientMaxWorkers(shoot *gardencorev1beta1.Shoot, reason string) []string {
+	var (
+		reasonsForUpdate []string
+	)
+
+	for i, worker := range shoot.Spec.Provider.Workers {
+		if int(worker.Maximum) >= len(worker.Zones) {
+			continue
+		}
+		newMaximum := int32(len(worker.Zones))
+		reasonsForUpdate = append(reasonsForUpdate, fmt.Sprintf("Maximum of worker-pool %q upgraded from %d to %d. Reason: %s", worker.Name, worker.Maximum, newMaximum, reason))
+		shoot.Spec.Provider.Workers[i].Maximum = newMaximum
+	}
```

Comment:
This check is only required for worker groups running system components, see:
- https://github.com/gardener/gardener/blob/bdfc06dc5cb4e5764800fd31ba1dd07727ad78bf/pkg/apis/core/validation/shoot.go#L1810
- https://github.com/gardener/gardener/blob/bdfc06dc5cb4e5764800fd31ba1dd07727ad78bf/pkg/apis/core/validation/shoot.go#L1847